assignment2/node.py

```diff
@@ -0,0 +1,15 @@
+class Node:
+    """Implementation of a Binary Trees node.
+
+    Fields:
+        key:   an integer, key of the current node;
+        left:  a Node, left child of the current node;
+        right: a Node, right child of the current node.
+    """
+    def __init__(self, key, left = None, right = None):
+        self.key = key
+        self.left = left
+        self.right = right
+
+    def __str__(self):
+        return self.key.__str__()
```

Comment:
Optional.

You can also write this as:
```python
  return str(self.key)
```
generally calling `__XXX__` methods is less preferable.